### ISSUE TITLE
Cover EKS certificate authority being undefined with a test

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-eks#904]
Reference:

- https://github.com/pulumi/pulumi-eks/issues/676
- https://github.com/pulumi/pulumi-eks/pull/903

Engineering was not able to add tests due to high cost of eks.Cluster ProgramTest execution and no viable alternatives to test the logic out.

**Comment by t0yv0** [comment]
Oh. Could you file that please? I'm not quite seeing it. You mean that a property that's marked "required output" is sometimes unknown?

**Comment by danielrbradley** [comment]
Yup that's what #676 is tracking. We still don't have a full repro so my PR was just fixing the immediate mishandling of undefined. However, there's still more investigation to do but my initial hypothesis points back to AWS in this comment: 
- https://github.com/pulumi/pulumi-eks/issues/676#issuecomment-1688231424

**Comment by t0yv0** [comment]
https://github.com/pulumi/pulumi-terraform-bridge/issues/1239 related perhaps. There's some more bridge issues I recall vaguely that don't guarantee that an output property is never nil.

**Comment by danielrbradley** [comment]
Yes that might be relevant @t0yv0 - it's a computed list property. It's not set as optional, but perhaps is always kinda optional given the list could have zero elements?

Here's where we set the value. I'm unsure how the empty map will get serialized .. will this be removed and therefore be undefined?

https://github.com/pulumi/terraform-provider-aws/blob/225e17c8a683de1264a1e6d07e2ccc62852dead4/internal/service/eks/cluster.go#L908-L918
